src/Site.js

```diff
@@ -940,13 +943,53 @@ Site.prototype.writeSiteData = function () {
   });
 };
 
+function constructDeploymentUrl(remoteRepo) {
+  const parts = remoteRepo.split('/');
+  let deploymentUrl = '';
+  if (parts[parts.length - 1].includes(GITHUB_IO_STRING)) {
+    // Use the github.io link itself
+    deploymentUrl = `https://${parts[parts.length - 1].substring(0, parts[parts.length - 1].length - 4)}`;
+  } else {
+    // Use the org_name + repo_name to reconstruct the url
+    deploymentUrl = `https://${parts[parts.length - 2]}.${GITHUB_IO_STRING}/`
+      + `${parts[parts.length - 1].substring(0, parts[parts.length - 1].length - 4)}`;
+  }
+  return deploymentUrl;
```

Comment:
GitHub allows custom domain through the use of `CNAME`. Check it out [here](https://help.github.com/en/github/working-with-github-pages/configuring-a-custom-domain-for-your-github-pages-site).

Therefore, this method would generate the wrong URL if the user is using a custom domain.

Reply:
I see. Currently looking into programmatically getting the `CNAME` that points to the default GitHub pages domain. :v: